activerecord/lib/arel/visitors/postgresql.rb

```diff
@@ -64,7 +64,7 @@ def prepare_update_statement(o)
             # The PostgreSQL dialect isn't flexible enough to allow anything other than a inner join
             # for the first join:
             #   UPDATE table SET .. FROM joined_table WHERE ...
-            (o.relation.right.all? { |join| join.is_a?(Arel::Nodes::InnerJoin) || join.right.expr.right.relation != o.relation.left })
+            (o.relation.right.all? { |join| join.is_a?(Arel::Nodes::InnerJoin) || join.right.expr.right.try(:relation) != o.relation.left })
```

Comment:
`try` is awful for performance, we shouldn't use that. We should check the type of the node instead.